### ISSUE TITLE
Learn Website

### ISSUE THREAD

**Issue by iros** [learntextvis/code-samples#2]
- [x] Set up infrastructure
- [x] gh-pages hosting for now
- [x] Build process + deployment
- [ ] Figure out embedding of live examples (iframes?)
- [ ] Basic code blocks - Highlight and format code samples (in the content)
- [x] Choose a writing format (markdown)

Nice to Have:
- Print stylesheet for PDF output or other format. (Keep in mind as we choose a build system.)

Content:
- Having some indexes of all visualizations & analysis topics
- Table of contents
- Placeholder pages
- Images directory
- Link to source
- Installation & set-up
- Unicode

**Comment by vlandham** [comment]
proposal: 

use jekyll for build system. it has all of these features and more. It is also a mature package and integrates well into gh-pages.

use pandoc for converting to pdf. 

### Structure:

New repo for learn website:

**learntextvis or learntextvis.com or leartextvis-site** inside the leartextvis org.

Sample content of repo:

```
_posts
  - 01-01-2016-chapter01-.md
_layouts
 - chapter_layout.html
_includes
 - footer.html
 - header.html

index.html
```

### Publishing

we can ensure our site needs no preprocessing and simply use a single branch **gh-pages**.

**Comment by vlandham** [comment]
proposal 2:

use [pelican](http://docs.getpelican.com/en/3.6.3/) for build system. It is written in python and has a bunch of interesting plugins: https://github.com/getpelican/pelican-plugins

**Comment by iros** [comment]
- We may need to also include iframe content somewhere, so it might make sense to have a separate folder for this, like `embeds` or something.
- What about any js we may want to have? even for basic page interactions?
- What about CSS? that has to live somewhere too. This requires a _sass folder and also a css folder with a main that imports all the _sass files. It's weird.  http://jekyllrb.com/docs/assets/ 
- Jekyll runs a build by default into a _site folder. I think separating source from build is important so I would suggest moving all the jekyll source files into a `src` dir and have a build folder separately (that we would pipe to gh-pages.)  It's pretty [easy](https://github.com/bocoup/OpenVisConf.com/blob/master/_config.yml) to do.
  To factor these in:
- We need a place for images and data files

```
src/
  _posts/
  _layouts/
  _sass/
    - layout.scss
    - toc.scss
    - embeds.scss
    - mq.scss
  css/
    - main.scss
  assets/
    - img/
    - data/
  js/
    - main.js (etc)

build/
_config.yml
<your build instructions js or whatever files>
```

**Comment by vlandham** [comment]
yep. my start was just to show how "chapters" could be "posts" in the jekyll world. 

i will look at the sass stuff to see if it doesnt add too much in complexity.
